connman/src/inet.c

```diff
@@ -778,8 +778,11 @@ int connman_inet_add_ipv6_network_route_with_metric(int index, const char *host,
 
 	rt.rtmsg_flags = RTF_UP;
 
-	/* Add host flag for route only when not setting a default route. */
-	if (!__connman_inet_is_any_addr(host, AF_INET6))
+	/*
+	 * Add host flag for route only when not setting a default route and
+	 * the route is not to a single IPv6 host address.
+	 */
+	if (!__connman_inet_is_any_addr(host, AF_INET6) && metric < 128)
```

Comment:
Because of this comment I looked into this. Search engines are a complete bust these days. But ChatGPT may be more useful:

Me:
> So the RTF_HOST must be set only for an IPv6 route that points to a /128 prefixed host and it is not the default route when adding a route with struct in6_rtmsg sent with ioctl() to the Linux kernel?
Yes, that is correct. The RTF_HOST flag should only be set for an IPv6 route that points to a /128 prefixed host. This flag indicates that the route is a host route and should be treated differently than routes with larger prefix lengths. In particular, it indicates that the destination address of the route is a specific host and not a network or subnet.

ChatGPT:
> When adding a route to the Linux kernel using the ioctl() system call and the struct in6_rtmsg structure, the RTF_HOST flag should only be set if the destination address has a prefix length of 128. If the destination address has a larger prefix length, then the RTF_HOST flag should not be set.

I guess I need to change this to use the prefix lenght instead of metric:
`if (!__connman_inet_is_any_addr(host, AF_INET6) && prefix_len == 128)`